.github/workflows/push.yml

```diff
@@ -0,0 +1,53 @@
+# This workflow will do a clean installation of node dependencies, cache/restore them, and run tests across different versions of node
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-nodejs
+# See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+
+name: apps/super-app CI (push to master)
+
+on:
+  push:
+    branches: [ "master" ]
+
+jobs:
+  build_and_deploy:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: 'read'
+      id-token: 'write'
+    strategy:
+      matrix:
+        node-version: [18.x]
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+        cache: 'npm'
+        cache-dependency-path: apps/super-app/package-lock.json
+    - name: Install
+      run: npm ci
+      working-directory: apps/super-app/
+    - name: Lint
+      run: npm run lint
+      working-directory: apps/super-app/
+    - name: Check formatting
+      run: npm run format:check
+      working-directory: apps/super-app/
+    - name: Test
+      run: npm test
+      working-directory: apps/super-app/
+    - name: Build
+      run: npm run build
+      working-directory: apps/super-app/
+    - name: 'Authenticate'
+      uses: 'google-github-actions/auth@v1'
+      with:
+        workload_identity_provider: 'projects/427853968081/locations/global/workloadIdentityPools/github-actions-pool/providers/github-actions-provider'
```

Comment:
Could we add a TODO to add this to the terraform setup?

Reply:
Done in [eae7305](https://github.com/bettersg/call-home/pull/167/commits/eae7305fa575ec5167b5c6c2027334737095741d)